### ISSUE TITLE
Error: @prisma/client did not initialize yet, another variant

### ISSUE THREAD

**Issue by jdevng** [prisma/prisma#13289]
### Bug description

This server has been working for a year with apollo, and last week I've been attempting to upgrade the libraries. This client error is well experienced on SO, but none of the solutions appear to work. The database is migrated properly, tables are present as expected.

### How to reproduce

1. Go to server directory
2. Change package.json as given in env
3. Run the following:

npm install
cd prisma
npx prisma migrate reset --preview-feature
npx prisma migrate dev --name init --preview-feature
npx prisma generate
npm run dev

4. See Error

/server/node_modules/.prisma/client/index.js:1
Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues
    at new PrismaClient (/server/node_modules/.prisma/client/index.js:3:11)
    at Object.<anonymous> (/server_broke/src/services/accounts/index.js:11:19)
    at Generator.next (<anonymous>)
    at bl (/server/node_modules/esm/esm.js:1)
    at kl (/server/node_modules/esm/esm.js:1)
    at Object.u (/server/node_modules/esm/esm.js:1)
    at Object.o (/server/node_modules/esm/esm.js:1)
    at Object.<anonymous> (/server/node_modules/esm/esm.js:1)
    at Object.apply (/server/node_modules/esm/esm.js:1)
    at /server/node_modules/esm/esm.js:1

Node.js v17.3.0

### Expected behavior

Expect prisma client to startup.

### Prisma information

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
  // previewFeatures = ["nativeTypes"]
}

model Account{
    id              Int             @default(autoincrement()) @id
    auth0           String          @default("")
    first_name       String          @default("")
    last_name        String          @default("")
    email           String
    created_at       DateTime        @default(now())
}

### Environment & setup

- OS: 
- Mac OS
- Database: 
-MySQL - but it's not starting up
- Node.js version: 
% node -v
v17.3.0

{
  "name": "apollo_server",
  "version": "1.0.0",
  "main": "index.js",
  "scripts": {
    "dev": "concurrently -k npm:dev:*",
    "dev:accounts": "nodemon -r dotenv/config -r esm ./src/services/accounts/index.js",
    "dev:gateway": "wait-on tcp:4001 && nodemon -r dotenv/config -r esm ./src/index.js",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "prisma": {
    "schema": "prisma/schema.prisma"
  },
  "dependencies": {
    "@apollo/federation": "^0.36.1",
    "@apollo/gateway": "^2.0.2",
    "@prisma/client": "^3.13.0",
    "apollo-server": "^3.7.0",
    "apollo-server-express": "^3.7.0",
    "auth0": "^2.40.0",
    "dotenv": "^16.0.0",
    "esm": "^3.2.25",
    "express": "^4.17.1",
    "express-jwt": "^7.7.0",
    "graphql": "^16.5.0",
    "graphql-middleware": "^6.1.26",
    "graphql-shield": "^7.4.2",
    "jwks-rsa": "^2.1.1",
    "jwt-decode": "^3.1.2",
    "nodemon": "^2.0.16",
    "sjcl": "^1.0.8",
    "wait-on": "^6.0.1"
  },
  "devDependencies": {
    "concurrently": "^7.1.0",
    "prisma": "^3.13.0"
  },
  "description": ""
}

### Prisma Version

```

```

3.13.0

**Comment by wesreid** [comment]
Same issue with Prisma version 3.14.0
`Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues`

In my case, local (MacOS/Darwin) works fine, but deployment to AWS Lambda (RHEL) fails with the above message when instantiating the client.

I was forced to revert to a previously-working AWS Lambda deployment Prisma version.
Deploying the same code to AWS Lambda using Prisma v3.8.1 works without issue.

**Comment by jdevng** [comment]
> Same issue with Prisma version 3.14.0 `Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again. In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues`
> 
> In my case, local (MacOS/Darwin) works fine, but deployment to AWS Lambda (RHEL) fails with the above message when instantiating the client.
> 

You were the first pioneer. Do you think it's a race condition for some reason, maybe the client is creating a new thread that didn't exist previously?

**Comment by glarivie** [comment]
Same with 3.11

**Comment by jdevng** [comment]
How low do we need to go? :)

**Comment by janpio** [comment]
@jdevng What is the output when you run `npx prisma generate`?
Does it actually change anything in `node_modules/.prisma/client`?

@wesreid @glarivie Please open your own bug report issues so we can investigate this. Thanks.

**Comment by jdevng** [comment]
I believe I found the issue - when you have a directory structure of

server_app/
server_app/prisma

each with node_modules in them, if generate is run in server_app, it will update the directory below, but not the directory you are in. Is this expected behavior?

**Comment by janpio** [comment]
Can you elaborate on this a bit? Not sure I am getting it.

**Comment by janpio** [comment]
Generally the Client should always be generated in the _expected_ location, as that is definitely not the case for you I think this is problematic. Can you maybe create a repository with the structure you are describing and where exactly to run that command? That might be helpful.

**Comment by jdevng** [comment]
Yes, this is how I came across this issue. I removed the node_modules directory under prisma and the correct client was written.

**Comment by iconifyit** [comment]
I was able to fix this issue by changing the generator output location outside of the prisma directory. In my case I set it to `../generated`. I _think_ the reason this worked was because of the duplicate `node_modules` directories. I just bypassed the issue altogether.

`generator client {
  provider = "prisma-client-js"
  previewFeatures = ["fullTextSearch"]
  output = "../generated/client"
}`

**Comment by millsp** [comment]
@jdevng I checked the code and it seems that we will generate near where the schema is, if we find a client or the cli also installed there. Because you installed the CLI in `server/prisma` we probably also automatically installed `@prisma/client` for you when you ran generate for the first time. We should maybe relax that, and check that a `@prisma/client` could be installed higher on the file system tree.

**Comment by jdevng** [comment]
The previous comments and I believe some documents that show the 'client should always be generated in the "expected" location' really is a requirement that is asking for bugs, due to human bias.

In this case, since there are two options (or more), I would say the proper operation would be indicate an error, because only one client should be generated at a time. Or did I bump into some use case I'm not aware of for this?

**Comment by millsp** [comment]
@jdevng I'd be very grateful if you could provide us with a reproduction repository.

**Comment by jdevng** [comment]
I'll do it as soon as i can.

**Comment by eznix86** [comment]
Vercel is spitting this:

```sh
Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
--
21:24:46.547 | In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues
21:24:46.547 | at new PrismaClient (/vercel/path0/node_modules/.prisma/client/index.js:43:11)
21:24:46.547 | at 3076 (/vercel/path0/.next/server/chunks/6492.js:19:14)
21:24:46.547 | at __webpack_require__ (/vercel/path0/.next/server/webpack-runtime.js:25:43)
21:24:46.547 | at 16492 (/vercel/path0/.next/server/chunks/6492.js:43:72)
21:24:46.547 | at __webpack_require__ (/vercel/path0/.next/server/webpack-runtime.js:25:43)
21:24:46.547 | at 1516 (/vercel/path0/.next/server/app/api/auth/connect/route.js:122:12)
21:24:46.547 | at __webpack_require__ (/vercel/path0/.next/server/webpack-runtime.js:25:43)
21:24:46.547 | at __webpack_exec__ (/vercel/path0/.next/server/app/api/auth/connect/route.js:236:39)
21:24:46.547 | at /vercel/path0/.next/server/app/api/auth/connect/route.js:237:103
21:24:46.547 | at __webpack_require__.X (/vercel/path0/.next/server/webpack-runtime.js:150:21)
21:24:46.550 |  
21:24:46.551 | > Build error occurred
21:24:46.553 | Error: Failed to collect page data for /api/auth/connect
21:24:46.554 | at /vercel/path0/node_modules/next/dist/build/utils.js:1161:15
21:24:46.554 | at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
21:24:46.554 | type: 'Error'
21:24:46.554 | }
21:24:46.652 | error Command failed with exit code 1.
21:24:46.652 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
21:24:46.673 | Error: Command "yarn run build" exited with 1
```

**Comment by janpio** [comment]
Your app seems to be importing from `@prisma/client` or `.prisma/client` @eznix86. When you update your import to the custom path you have, it should work.

**Comment by eznix86** [comment]
> Your app seems to be importing from `@prisma/client` or `.prisma/client` @eznix86. When you update your import to the custom path you have, it should work.

@janpio Thanks -> https://youtu.be/8aRCmMlCcY0

TLDR; (for other devs) I was importing types from `@prisma/client`, import your types from your prisma output parth. Me I use aliases, I just do:

```typescript
import type { User } from "@/generated/client" // in your case is the path you mentionned
```

**Comment by yayoamigo** [comment]
> > Your app seems to be importing from `@prisma/client` or `.prisma/client` @eznix86. When you update your import to the custom path you have, it should work.
> 
> @janpio Thanks -> https://youtu.be/8aRCmMlCcY0
> 
> TLDR; (for other devs) I was importing types from `@prisma/client`, import your types from your prisma output path if you have a custom one. Me I use aliases, I just do:
> 
> ```typescript
> import type { User } from "@/generated/client" // in your case is the path you mentioned
> ```
I was having the same problem thank you so much
